include/cgimap/xml_writer.hpp

```diff
@@ -38,24 +42,40 @@ class xml_writer : public output_writer {
   ~xml_writer() noexcept override;
 
   // begin a new element with the given name
-  void start(const std::string &name);
+  void start(const char *name);
+  inline void start(const std::string &name) { start(name.c_str()); }
 
   // write an attribute of the form name="value" to the current element
-  void attribute(const std::string &name, const std::string &value);
+  void attribute(const char *name, const std::string &value);
 
   // write a mysql string, which can be null
-  void attribute(const std::string &name, const char *value);
+  void attribute(const char *name, const char *value);
 
   // overloaded versions of writeAttribute for convenience
-  void attribute(const std::string &name, double value);
-  void attribute(const std::string &name, int32_t value);
-  void attribute(const std::string &name, int64_t value);
-  void attribute(const std::string &name, uint32_t value);
-  void attribute(const std::string &name, uint64_t value);
-  void attribute(const std::string &name, bool value);
+  void attribute(const char *name, double value);
+  void attribute(const char *name, bool value);
+
+  template<typename T>
+  inline void attribute(const std::string &name, T value) { attribute(name.c_str(), value); }
+
+  template<typename TInteger, std::enable_if_t<std::is_integral_v<TInteger>, bool> = true>
+  void attribute(const char *name, TInteger value) {
+    std::array<char, string_buf_size_v<TInteger>> buf;
```

Comment:
Could we maybe use some hardcoded value here instead, like 32? The array is residing on the stack only, and the memory isn't being initialized, so there shouldn't be much of an extra cost to have a larger buffer size. This way, we wouldn't need the additional logic in util.hpp. I believe this could be good enough for our purposes.

Reply:
That works until someone passes in a `__int128`. (It's probably not very likely that will ever happen, but who knows.)

The `static_assert`s here https://github.com/zerebubuth/openstreetmap-cgimap/pull/342/files#diff-64a14965469222c18245822fae47ea2eff593b71d3a397d19e773bff633fae62R154-R159 provide some sanity checks that the size logic works as intended.

(`static_assert(string_buf_size_v<__int128> == 45);` for `__int128`)